report/record.go

```diff
@@ -0,0 +1,92 @@
+package report
+
+import (
+	"io"
+	"sync"
+
+	"github.com/pterm/pterm"
+)
+
+var m = &sync.Mutex{}
+
+var records = make(map[string]Entry)
+
+func Record(e Entry) {
+	defer m.Unlock()
+	m.Lock()
+	records[e.Identifier] = e
+}
+
+// RecordBatch accepts a BatchEntry that contains a slice of identifiers, loops through them and converts each identifier to
+// a standard Entry. This is useful for supporting batch delete workflows in cloud-nuke (such as cloudwatch_dashboards)
+func RecordBatch(e BatchEntry) {
+	for _, identifier := range e.Identifiers {
+		entry := Entry{
+			Identifier:   identifier,
+			ResourceType: e.ResourceType,
+			Error:        e.Error,
+		}
+		Record(entry)
+	}
+}
+
+func Print(w io.Writer) {
+	renderSection("Nuking complete:", w)
+	data := make([][]string, len(records))
+	entriesToDisplay := []Entry{}
+	for _, entry := range records {
+		entriesToDisplay = append(entriesToDisplay, entry)
+	}
+
+	for idx, entry := range entriesToDisplay {
+		var errSymbol string
+		if entry.Error != nil {
+			errSymbol = "     ❌    "
+		} else {
+			errSymbol = "     ✅    "
+		}
+		data[idx] = []string{entry.Identifier, entry.ResourceType, errSymbol}
+	}
+
+	renderTableWithHeader([]string{"Identifier", "Resource Type", "Deleted Successfully"}, data, w)
+}
+
+func renderSection(sectionTitle string, w io.Writer) {
+	section := pterm.DefaultSection.WithStyle(pterm.NewStyle(pterm.FgLightCyan))
+	section = section.WithWriter(w).WithLevel(0)
+	section.Println(sectionTitle)
+}
+
+func renderTableWithHeader(headers []string, data [][]string, w io.Writer) {
+	tableData := pterm.TableData{
+		headers,
+	}
+	for idx := range data {
+		tableData = append(tableData, data[idx])
+	}
+	pterm.DefaultTable.
+		WithHasHeader().
+		WithBoxed(true).
+		WithRowSeparator("-").
+		WithData(tableData).
+		WithWriter(w).
+		Render()
```

Comment:
Unchecked returned error from `Render()` function

Reply:
Ah, thank you good catch.

Reply:
Fixed in [304ac6c](https://github.com/gruntwork-io/cloud-nuke/pull/356/commits/304ac6c19eb2531fa9d77ba9753756241cbc314a)